<launch>

	<include file="$(find lab4)/launch/MapServer.launch"/>

	<!-- bag_path: The path to get a pose of the car for Sensor Model simulation
         scan_topic: The topic containing laser scans
		 laser_ray_step: Step for downsampling laser scans. 
		 Ideally: 720 ray samples / 20 = 36 downsamples. 360 degrees / 36 downsamples = 10 degree step
		 Reality: Number of laser scan samples could be any less than 720 
		 exclude_max_range_rays: Whether to exclude rays that are beyond the max range
		 max_range_meters: The max range of the laser. /home/robot/catkin_ws/src/mushr_sim/config/sensors.yaml file default = 10.0. TAs default = 5.6 -->
  	<arg name="bag_path" default="/home/robot/tutorial_ws/src/lab4/bags/laser_scans/laser_scan3.bag"/>
  	<arg name="scan_topic" default="/scan"/>
	<arg name="laser_ray_step" default="20"/>
	<arg name="exclude_max_range_rays" default="True"/>
	<arg name="max_range_meters" default="10.0"/>
	
	<node pkg="lab4" type="SensorModel.py" name="sensor_model" output="screen">
		<param name="bag_path" type="string" value="$(arg bag_path)"/>
		<param name="scan_topic" type="string" value="$(arg scan_topic)"/>
		<param name="laser_ray_step" type="int" value="$(arg laser_ray_step)"/>
		<param name="exclude_max_range_rays" type="bool" value="$(arg exclude_max_range_rays)"/>
		<param name="max_range_meters" type="double" value="$(arg max_range_meters)"/>
	</node>

</launch>
